docs/setup/independent/high-availability.md

```diff
@@ -0,0 +1,508 @@
+        name: etcd
+        env:
+        - name: PUBLIC_IP
+        valueFrom:
+            fieldRef:
+            fieldPath: status.hostIP
+        - name: PRIVATE_IP
+        valueFrom:
+            fieldRef:
+            fieldPath: status.podIP
+        - name: PEER_NAME
+        valueFrom:
+            fieldRef:
+            fieldPath: metadata.name
+        volumeMounts:
+        - mountPath: /var/lib/etcd
+        name: etcd
+        - mountPath: /certs
+        name: certs
+    hostNetwork: true
+    volumes:
+    - hostPath:
+        path: /var/lib/etcd
+        type: DirectoryOrCreate
+        name: etcd
+    - hostPath:
+        path: /etc/kubernetes/pki/etcd
+        name: certs
+    EOL
+    ```
+
+    Make sure you replace:
+    * `<podname>` with the name of the node you're running on (e.g. `etcd0`, `etcd1` or `etcd2`)
+    * `<etcd0-ip-address>`, `<etcd1-ip-address>` and `<etcd2-ip-address>` with the public IPv4s of the other machines that host etcd.
+
+{% endcapture %}
+
+{% assign tab_names = "Choose one...,systemd,Static Pods" | split: ',' | compact %}
+{% assign tab_contents = site.emptyArray | push: choose | push: systemd | push: static_pods %}
+
+{% include tabs.md %}
+
+## Set up master Load Balancer
+
+The next step is to create a Load Balancer that sits in front of your master nodes. How you do this depends on your environment; you could, for example, leverage a cloud provider Load Balancer, or set up your own using nginx, keepalived, or HAproxy. Some examples of cloud provider solutions are:
+
+* [AWS Elastic Load Balancer](https://aws.amazon.com/elasticloadbalancing/)
+* [GCE Load Balancing](https://cloud.google.com/compute/docs/load-balancing/)
+* [Azure](https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-overview)
+
+You will need to ensure that the load balancer routes to **just `master0` on port 6443**. This is because kubeadm will perform health checks using the load balancer IP. Since `master0` is set up individually first, the other masters will not have running apiservers, which will result in kubeadm hanging indefinitely.
+
+If possible, use a smart load balancing algorithm like "least connections", and use health checks so unhealthy nodes can be removed from circulation. Most providers will provide these features.
+
+## Acquire etcd certs
+
+Only follow this step if your etcd is hosted on dedicated nodes (**Option 1**). If you are hosting etcd on the masters (**Option 2**), you can skip this step since you've already generated the etcd certificates on the masters.
+
+1. Generate SSH keys for each of the master nodes by following the steps in the [create ssh access](#create-ssh-access) section. After doing this, each master will have an SSH key in `~/.ssh/id_rsa.pub` and an entry in `etcd0`'s `~/.ssh/authorized_keys` file.
+
+1. Run the following:
+
+    ```shell
+    mkdir -p /etc/kubernetes/pki/etcd
+    scp root@<etcd0-ip-address>:/etc/kubernetes/pki/etcd/ca.pem /etc/kubernetes/pki/etcd
```

Comment:
@jamiehannaford, I noticed that you don't copy the _etcd_ CA key across here. I'm finding today that I can't get this to work adequately, per kubernetes/kubeadm#807. Did this used to work as of late 2017?

Reply:
@seh fix is in flight for that issue.  /cc @detiber

Reply:
https://github.com/kubernetes/kubernetes/pull/63806